addons/hr_contract/models/hr_employee.py

```diff
@@ -97,6 +99,12 @@ def write(self, vals):
                 employee.resource_calendar_id.transfer_leaves_to(employee.contract_id.resource_calendar_id, employee.resource_id)
                 employee.resource_calendar_id = employee.contract_id.resource_calendar_id
         return res
+    
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_open_contract(self):
+        for employee in self:
+            if employee.contract_ids.filtered(lambda contract: contract.state == 'open'):
+                raise UserError(_('You cannot delete an employee with a running contract.'))
```

Comment:
Filter is inefficient, because you iterate trough all your contracts; so it's like a double loop. :D
Your goal is to find at least one employee with a running contract so you can do it:
```suggestion
        if any(contract.state == 'open' for contract in self.contract_ids)
                raise UserError(_('You cannot delete an employee with a running contract.'))
```